lib/Net/Async/Blockchain/ETH.pm

```diff
@@ -77,6 +78,43 @@ An hexadecimal string
 
 sub subscription_id { shift->{subscription_id} }
 
+=head2 accounts
+
+return the blockchain accounts, if is the first time will call eth_accounts.
+
+=over 4
+
+=back
+
+returns a Future, the on_done response will be the accounts array.
+
+=cut
+
+async sub accounts {
+    my $self = shift;
+    return $self->{accounts} //= do {
+        await $self->rpc_client->accounts();
+    };
+}
+
+=head2 update_accounts
+
+update the C<accounts> variable every 10 seconds
+
+=over 4
+
+=back
+
+=cut
+
+async sub update_accounts {
+    my $self = shift;
+    while (1) {
```

Comment:
This never terminates?

Reply:
It needs to be live while the subscription is active, basically all the time.